test/base58_encode_test.exs

```diff
@@ -0,0 +1,38 @@
+defmodule Base58EncodeTest do
+  use ExUnit.Case
+  use ExUnitProperties
+
+  doctest Base58Encode
+
+  describe "Testing encode function" do
+    test "returns base58 for the string foo" do
+      assert "bQbp" == Base58Encode.encode("foo")
+    end
+
+    test "returns error if parameter is not a binary" do
+      assert :error == Base58Encode.encode(23)
+    end
+
+    test "returns z when binary is represented by 57" do
+      assert "z" == Base58Encode.encode(<<57>>)
+    end
+
+    test "returns empty string when encoding an empty string" do
+      assert "" == Base58Encode.encode("")
+    end
+
+    test "add leading zeros as \"1\"" do
+      assert "1112" == Base58Encode.encode(<<0, 0, 0, 1>>)
+    end
+
+    test "encode empty string" do
+      assert "Z" == Base58Encode.encode(" ")
+    end
+
+    property "Compare result with basefiftyeight package" do
```

Comment:
Ahhhh that's really clever. I was wondering what a good way to add property tests to this would be. Good thinking @SimonLab 👍 😎

Reply:
A good "stopgap" until we can implement #4 and test against the "reference" **`C`** version 
(_and avoid comparing to the unmaintained one ... unless we want to run a perf benchmark?_)

Reply:
see #4 for the next step on this (use the C library of base58 instead of the elixir package)